packages/server/src/cors.ts

```diff
@@ -0,0 +1,115 @@
+import z from 'zod';
+import { env } from './env/server';
+import { RunningEnv } from './env/schema';
+
+const corsWhitelist = [
+    '.nav.no',
+    '.oera.no',
+    '.nais.io',
+    'https://preview-sykdomifamilien.gtsb.io',
+    'navdialog.cs102.force.com',
+    'navdialog.cs106.force.com',
+    'navdialog.cs108.force.com',
+    'navdialog.cs162.force.com',
+];
+
+
+export const isAllowedDomain = (origin?: string) =>
+    origin && corsWhitelist.some((domain) => origin.endsWith(domain));
+
+export const corsSchema = z.string().refine(
+    (origin) => {
+        return origin.includes('localhost') || isAllowedDomain(origin);
+    },
+    (origin) => {
+        if (!origin) {
+            return {
+                message: 'Origin header is missing',
+            };
+        }
+
+        return {
+            message: `Origin ${origin} is not allowed. Valid origins are ${corsWhitelist.join(
+                ', '
+            )}`,
+        };
+    }
+);
+
+// At the start of each request
+type CorsError = {
+    kind: 'cors-error',
+    response: Response
+}
+
+type Valid = {
+    kind: 'valid',
+    headers: HeadersInit
+}
+type Result = CorsError | Valid;
+
+
+export function handleCors(request: Request): Result {
+    const host = request.headers.get('host');
+    const result = corsSchema.safeParse(host);
+
+    if (!result.success) {
+        return {
+            kind: 'cors-error' as const,
+            response: new Response(null, {
+                status: 403,
+            })
+        } as const
+    }
+
+    const origin = result.data;
+
+    const shared = {
+        'Access-Control-Allow-Methods': 'GET,HEAD,OPTIONS,POST,PUT',
+    }
+
+
+    const headers: Record<RunningEnv, HeadersInit> = {
```

Comment:
Gjorde det på denne måten for å ikke lage en unødvendig abstraksjon på toppen av Response. Så merges disse inn i Responsen i rquest handler.